crates/data_components/src/graphql/provider.rs

```diff
@@ -27,19 +27,39 @@ use datafusion::{
 };
 use std::{any::Any, sync::Arc};
 
-use super::client::GraphQLClient;
 use super::Result;
+use super::{client::GraphQLClient, ResultTransformSnafu};
+
+pub type TransformFn = Arc<
+    dyn Fn(&RecordBatch) -> Result<RecordBatch, Box<dyn std::error::Error + Send + Sync>>
+        + Send
+        + Sync,
+>;
 
 pub struct GraphQLTableProvider {
     client: GraphQLClient,
-    schema: SchemaRef,
+    gql_schema: SchemaRef,
+    table_schema: SchemaRef,
+    transform_fn: Option<TransformFn>,
 }
 
 impl GraphQLTableProvider {
-    pub async fn new(client: GraphQLClient) -> Result<Self> {
-        let (_, schema, _) = client.execute(None, None, None).await?;
+    pub async fn new(client: GraphQLClient, transform_fn: Option<TransformFn>) -> Result<Self> {
```

Comment:
Let's change this to not require `transform_fn` but instead is set with a builder-like fn, i.e. `fn with_schema_transform(mut self, ...`

Reply:
Refactored to use builder pattern